### PR TITLE
Add random forest option for partial dependence

### PR DESCRIPTION
**Pull request by katiedagon**
@djgagne @jsschreck Following our discussion on the MILES slack about the behavior of XGBoost sometimes generating negative values in partial dependence plots, I've attempted to update the code to add a random forest option. I set it up to use RF as the default, but that could be changed. I tested this with an existing echo run, and the PD plots are comparable to those produced with XGB, but still generating negative values which I'm not sure makes sense for these metrics. [Here](https://docs.google.com/presentation/d/1noQLFlN6g9eFbbdV-n0sL5LhApoFrZ7ExXFW5dPJB5U/edit?usp=sharing) are some example comparisons using a set of metrics from @mariajmolina - noting the vcorr_cust metric is likely out of date, but the valid_mae and vmse_extreme_outp should be relevant (and I believe should not have negative values).

I welcome comments/suggestions on any of the above, plus how best to let users choose between the two options, in addition to general comments about how the RF model is configured (e.g., hyperparameters).

### CODE REVIEW

File: echo/src/partial_dependence.py [hunk 1]
```diff
@@ -56,41 +58,62 @@ def partial_dep(fn, input_cols, output_col, verbose=0):
     y_valid = yscaler.transform(np.expand_dims(y_valid, -1))
```
Comment (djgagne) [inline echo/src/partial_dependence.py:58]: I would remove the yscaler since random forest and XG Boost do not need their input or output scaled to work well.

File: echo/src/partial_dependence.py [hunk 2]
```diff
@@ -27,6 +28,7 @@ def partial_dep(fn, input_cols, output_col, verbose=0):
             df[param] = le.fit_transform(df[param])
 
     objective = "reg:squarederror"
```
Comment (djgagne) [inline echo/src/partial_dependence.py:30]: Instead of hard coding the model hyperparameters, I would put them in a dictionary called hparams and make that a keyword argument for partial_dep.

**Review by djgagne — CHANGES_REQUESTED** [review]
@katiedagon Thank you for adding the random forest functionality. I added a couple of small suggestions that should address the remaining issues with negative output scores.

File: echo/src/partial_dependence.py (hunk 1 above)
Comment (katiedagon) [reply echo/src/partial_dependence.py:58]: Ok, I will comment out yscaler lines similar to xscaler.

File: echo/src/partial_dependence.py (hunk 2 above)
Comment (katiedagon) [reply echo/src/partial_dependence.py:30]: Sounds good - where should I define hparams?

File: echo/src/partial_dependence.py (hunk 1 above)
Comment (katiedagon) [reply echo/src/partial_dependence.py:58]: Hmmm...commenting out the yscaler lines did not address the negative output scores. I wonder if there is some additional scaling happening in the `partial_dependence_plot` function?

Comment (katiedagon) [reply echo/src/partial_dependence.py:58]: Oops never mind, checking the wrong file. It worked!

**Review by djgagne — APPROVED** [review]
Changes look good to me. Merging.
